storage/libradb/src/lib.rs

```diff
@@ -673,6 +673,17 @@ impl DbReader for LibraDB {
 
         Ok(tree_state)
     }
+
+    fn get_block_timestamp(&self, version: u64) -> Result<u64> {
+        // rely on the fact that blockmetadata txn always precedes user txn.
+        for i in (1..=version).rev() {
```

Comment:
Let's give it a bound by introducing a `MAX_VERSIONS_TO_SEARCH_FOR_BLOCK_META`, or similar.

Reply:
done.